File: LinkedList/linkedList.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert_at_head(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node

    def insert_at_end(self, data):
        new_node = Node(data)
        new_node.next = None

        if self.head is None:
            self.head = new_node
        else:
            current = self.head
            while current.next is not None:
                current = current.next
            current.next = new_node

    def delete_node(self, data):
        if self.head is None:
            return

        if self.head.data == data:
            temp = self.head
            self.head = self.head.next
            del temp
            return

        current = self.head
        while current.next is not None and current.next.data != data:
            current = current.next

        if current.next is None:
            return

        temp = current.next
        current.next = current.next.next
        del temp

    def print_list(self):
        current = self.head
        output = "Linked List: "
        while current is not None:
            output += str(current.data) + " "
            current = current.next
        print(output)

# Usage example
list = LinkedList()

list.insert_at_head(10)
list.insert_at_end(20)
list.insert_at_end(30)
list.insert_at_head(40)
list.delete_node(30)
list.delete_node(40)
list.insert_at_head(5)
list.print_list()

File: LinkedList/linkedList.go
package main

import "fmt"

type Node struct {
    data int
    next *Node
}

type LinkedList struct {
    head *Node
}

func (list *LinkedList) insertAtHead(data int) {
    newNode := &Node{data: data, next: list.head}
    list.head = newNode
}

func (list *LinkedList) insertAtEnd(data int) {
    newNode := &Node{data: data, next: nil}

    if list.head == nil {
        list.head = newNode
    } else {
        current := list.head
        for current.next != nil {
            current = current.next
        }
        current.next = newNode
    }
}

func (list *LinkedList) deleteNode(data int) {
    if list.head == nil {
        return
    }

    if list.head.data == data {
        list.head = list.head.next
        return
    }

    current := list.head
    for current.next != nil && current.next.data != data {
        current = current.next
    }

    if current.next == nil {
        return
    }

    current.next = current.next.next
}

func (list *LinkedList) printList() {
    current := list.head
    fmt.Print("Linked List: ")
    for current != nil {
        fmt.Print(current.data, " ")
        current = current.next
    }
    fmt.Println()
}

func main() {
    list := &LinkedList{}

    list.insertAtHead(10)
    list.insertAtEnd(20)
    list.insertAtEnd(30)
    list.insertAtHead(40)
    list.deleteNode(30)
    list.deleteNode(40)
    list.insertAtHead(5)
    list.printList()
}
